frontend/apps/reader/modules/readerbookmark.lua

```diff
@@ -292,56 +366,262 @@ function ReaderBookmark:onShowBookmark()
+            while s do
+                if s ~= 1 or cap ~= "" then
+                    table.insert(t, cap)
+                end
+                last_end = e + 1
+                s, e, cap = str:find(fpat, last_end)
+            end
+            if last_end <= #str then
+                cap = str:sub(last_end)
+                table.insert(t, cap)
+            end
+            return t
+        end
+
+        -- HELPER FUNCTIONS WHICH PROBABLY SHOULD BE PUT SOMEWHERE ELSE:
+
+        local function substrCount(subject, needle)
+            return select(2, subject:gsub(needle, ""))
+        end
+
+        -- put uppercase strings (authors etc.) at start of line, and always a linebreak between them and the following text:
+        local function uppercaseWordsAtStartOfLine(itext)
+            itext = string.gsub(itext, "\n[ ]+([A-Z][A-Z]+)", "\n%1")
+            itext = string.gsub(itext, "([A-Z]+)\n[ ]+", "%1\n\n")
+            itext = string.gsub(itext, "([A-Z]+)\n\n[ ]+", "%1\n\n")
+            itext = string.gsub(itext, "([A-Z]+)\n([A-Z])", "%1\n\n%2")
+            itext = string.gsub(itext, "\n\n\n", "\n\n")
+            return itext
+        end
+        local function isPoem(itext)
+            local line_endings_count = substrCount(text, "\n")
+            local requisition1 = line_endings_count > 3
+
+            -- check whether lines are not too long:
+            local lines = split(itext, "\n")
+            local requisition2 = true
+            for _, line in ipairs(lines) do
+                if string.len(line) > 52 then
+                    requisition2 = false
+                    break
+                end
+            end
+            return (requisition1 == true and requisition2 == true)
+        end
+        local function indent(itext)
+            -- only for non poetic text indent para's:
+            if not isPoem(itext) then
+                local paras = split(itext, "\n")
+                local skip_next_para = false
+                for nr, para in ipairs(paras) do
+                    if nr > 1 and para:match("[A-Za-z]") then
+                        if not skip_next_para then
+                            paras[nr] = "     " .. para
+                        else
+                            skip_next_para = false
+                        end
+                    elseif nr > 1 then
+                        skip_next_para = true
+                    end
+                end
+                itext = table.concat(paras, "\n")
+            end
+            return uppercaseWordsAtStartOfLine(itext)
+        end
+
         self.textviewer = TextViewer:new{
-            title = _("Bookmark details"),
-            text = item.notes,
+            title = title,
+            -- add indentation for better readability:
+            text = indent(text),
+            -- make text easier to read:
+            justified = false,
+            text_face = Font:getFace("xx_smallinfofont"),
```

Comment:
It looks like you reduced the font size from x_smallinfofont to xx_smallinfofont.
May be it's nicer, may be it's not, I don't know.
May be it's just because _you_ usually highlight huge text segments - but people who highlight smaller segments will feel robed of their readability :)
May be you could re-use the font size a user has set for dictionary/wikipedia results (even if it's unrelated, it might be more the size a user expect to read such text with than your hardcoded smaller one):
https://github.com/koreader/koreader/blob/c28f1b3450b9db073a072255a247744b5afb20ed/frontend/ui/widget/dictquicklookup.lua#L75-L77